docs/installation.md

```diff
@@ -1,30 +1,162 @@
 # Installing Kusk Gateway
 
-## Helm
-TODO
+## Helm installation
 
-## Locally
-If you want to run kusk-gateway locally, you can do this easily using Minikube
+We provide [Helm v3](https://helm.sh/) [charts](https://github.com/kubeshop/helm-charts) for the Kusk Gateway installation.
+
+There are 2 charts to install:
+
+* **[kusk-gateway](https://github.com/kubeshop/helm-charts/tree/main/charts/kusk-gateway)** chart provides Custom Resources Definitions as well as Kusk Gateway Manager (Operator) deployment.
+
+* **[kusk-gateway-envoyfleet](https://github.com/kubeshop/helm-charts/tree/main/charts/kusk-gateway-envoyfleet)** chart provides the EnvoyFleet Custom Resource installation, which is used to configure the gateway with KGW Manager.
+
+Container images are hosted on Docker Hub [Kusk-Gateway repository](https://hub.docker.com/r/kubeshop/kusk-gateway).
+
+### Prerequsities
+
+* [Helm v3](https://helm.sh/) and [Kubectl](https://kubernetes.io/docs/tasks/tools/) installed
+
+* Kubernetes cluster administration rights are required - we install CRDs, service account with ClusterRoles and RoleBindings.
+
+* We heavily depend on [jetstack cert-manager](https://github.com/jetstack/cert-manager) for webhooks TLS configuration. If it is not installed in your cluster, then please install it with the official instructions [here](https://cert-manager.io/docs/installation/).
+
+* If you try to install the chart to your local machine cluster (k3d or minikube), you may need to install and configure [MetalLB](https://metallb.universe.tf/) to handle LoadBalancer type services,
+otherwise EnvoyFleet service ExternalIP address will be in Pending state forever. See installing [Locally with Minikube and Helm](#locally-with-minikube-and-helm) 
+
+### Installation
+
+The commands below will install Kusk Gateway and the "default" Envoy Fleet (LoadBalancer) in the recommended **kusk-system** namespace.
+
+```sh
+helm repo add kubeshop https://kubeshop.github.io/helm-charts
+helm repo update
+helm install kusk-gateway kubeshop/kusk-gateway -n kusk-system --create-namespace
+kubectl wait --for=condition=available --timeout=600s deployment/kusk-gateway-manager  -n kusk-system
+helm install kusk-gateway-default-envoyfleet kubeshop/kusk-gateway-envoyfleet -n kusk-system
+```
+
+You can now deploy your Gateway Custom resources, see also Examples section below.
+
+### Uninstallation
+
+```sh
+helm delete kusk-gateway-default-envoyfleet kusk-gateway -n kusk-system && kubectl delete namespace kusk-system
+```
+
+## Locally with Minikube and Helm
+
+### Prerequisities
+
+Installed:
+
+* [Minikube](https://minikube.sigs.k8s.io/docs/)
+* [Helm v3](https://helm.sh/)
+* [Kubectl](https://kubernetes.io/docs/tasks/tools/)
+
+### Minikube cluster setup and installation with Helm
+
+Run the following set of commands to quickly setup Minikube-based K8s cluster and setup Kusk Gateway with Helm.
+
+```sh
+minikube start --profile kgw
```

Comment:
we need to specify that user either needs a `hyperkit` driver if he's on macOS or something else if he's on linux/windows.